src/components/viewer.ts

```diff
@@ -123,7 +124,7 @@ export class Viewer {
         // pdfjs viewer automatically call decodeURIComponent.
         // So, to pass the encoded path of a pdf file to the http server,
         // we have to call encodeURIComponent two times! 2 - 1 = 1 !
-        const url = `http://${this.extension.server.address}/viewer.html?incode=1&file=/pdf:${uri.authority ? `\\\\${uri.authority}` : ''}${encodeURIComponent(encodeURIComponent(uri.fsPath))}`
+        const url = `http://localhost:${this.extension.server.port}/viewer.html?incode=1&file=/pdf:${uri.authority ? `\\\\${uri.authority}` : ''}${encodeURIComponent(encodeURIComponent(uri.fsPath))}`
         return `
```

Comment:
I do not really get the difference between `${this.extension.server.address}` and `localhost:${this.extension.server.port}`but if there is indeed a difference should not it be changed also [line 76](https://github.com/James-Yu/LaTeX-Workshop/blob/78e070392db9052e95106bb50b91bbff6bd67598/src/components/viewer.ts#L76).

Reply:
This change is needed because VSCode treats `localhost` and `127.0.0.1` differently. In other words, `localhost` is hard-coded [here](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/contrib/webview/electron-browser/webviewElement.ts#L172). 

Reported in Microsoft/vscode-remote-release/issues/108. 

If WebView finds a request to `localhost:webviewPort`, it converts the request to `localhost:tunnelLocalPort` and send the request to a remote server via ssh tunneling, then on the remote server, the request is forwarded to `localhost:extensionHostPort`.  So, this change is necessary only on an html source for WebView. It is not needed in line 76, checkViewer.